q2_assembly/spades/spades.py

```diff
@@ -114,11 +124,36 @@ def _assemble_spades(seqs, meta, common_args) -> ContigSequencesDirFmt:
     manifest = seqs.manifest.view(pd.DataFrame)
     result = ContigSequencesDirFmt()
 
-    for samp in list(manifest.index):
-        fwd = manifest.loc[samp, "forward"]
-        rev = manifest.loc[samp, "reverse"] if paired else None
+    if coassemble:
+        print(
+            "WARNING: The co-assembly process includes concatenating contents of all"
+            "read files. This process may take a while to finish!"
+        )
+        fwds = manifest["forward"]  # this will be  a list
+        revs = (
+            manifest["reverse"] if paired else None
+        )  # this will be a list if it exists
+
+        extension = get_file_extension(fwds[0])
+        fwd = f"all_samples_fwd{extension}"
+        rev = f"all_samples_rev{extension}" if paired else None
+
+        concatenate_files(fwds, fwd)
+        if paired:
+            concatenate_files(revs, rev)
 
-        _process_sample(samp, fwd, rev, common_args, result)
+        _process_sample("all_samples_spades", fwd, rev, common_args, result)
```

Comment:
Please remove the "_spades" suffix - no need to indicate there that it was generated with metaspades (this info is preserved in the provenance).